pkg/ctl/update/cluster.go

```diff
@@ -72,6 +83,23 @@ func doUpdateClusterCmd(p *api.ProviderConfig, cfg *api.ClusterConfig, nameArg s
 		return err
 	}
 
+	if err := ctl.GetCredentials(cfg); err != nil {
+		return errors.Wrapf(err, "getting credentials for cluster %q", cfg.Metadata.Name)
+	}
+
+	currentVersion := ctl.ControlPlaneVersion()
+	switch currentVersion {
+	case "":
+		return fmt.Errorf("unable to get control plane version")
+	case api.Version1_10:
+		cfg.Metadata.Version = api.Version1_11
+	case api.LatestVersion:
+		cfg.Metadata.Version = api.LatestVersion
+	default:
+		return fmt.Errorf("control plane version version %q is not upgradable by this version of eksctl, try to upgrade eksctl first", currentVersion)
```

Comment:
What about also mentioning that eks might not support the version and to check the docs?

Reply:
👍

Reply:
Actually, this is about version you currently have, but something is not right about this code. I'll have a look at what it improving this before I merge the PR.

Reply:
I added comment and tweaked the error message.